File: 50.hs
-- Project Euler, Problem #50
-- Lucas Garron
-- Date: December 30, 2012

import Data.List (genericDrop, genericLength)
import Data.Numbers.Primes (isPrime, primes)

-- Code

limit :: Integer
limit = 10^6

primeSums :: [Integer]
primeSums = takeWhile (< limit) $ scanl1 (+) primes

partialprimeSums :: Integer -> [Integer]
partialprimeSums len = zipWith (-) (genericDrop len primeSums) primeSums

allowed :: Integer -> Bool
allowed x = x < limit && isPrime x

result :: Integer
result = last $ filter allowed $ concat $ map partialprimeSums [1..genericLength primeSums]

-- Output

main :: IO ()
main = print result

File: 50.py
# Project Euler, Problem #50
# Lucas Garron
# Date: November 26, 2013

from __future__ import print_function
import itertools
import pyprimes

MAX = 10 ** 6

primes = list(itertools.takewhile(lambda x: x < MAX, pyprimes.primes()))

# No itertools.accumulate in Python 2...
partialSums = []
partialSum = 0
for p in primes:
  partialSum += p
  partialSums.append(partialSum)

numPrimes = len(primes)

recordLength = 0
record = None

for i in range(numPrimes):
  for j in range(i, numPrimes):
    d = partialSums[j] - partialSums[i]
    if d < MAX and pyprimes.isprime(d):
      if j - i > recordLength:
        recordLength = j - i
        record = d
        print(recordLength, record)
